.circleci/config.yml

```
@@ -1,7 +1,135 @@
+      - run:
+          name: Install docker-compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.12.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+            chmod +x /usr/local/bin/docker-compose
+      - run:
+          name: Run Docker compose
+          command: |
+            set -x
+            docker-compose -f docker-compose-prod.yaml up -d
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+  install-dev:
+    docker:
+      - image: uwbpparamedics/uw-blueprint-paramedics-react:0.0.2
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+              - v1-dependencies-
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+  build-dev:
+    docker:
+      - image: uwbpparamedics/uw-blueprint-paramedics-react:0.0.2
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+      - node/with-cache:
+          steps:
+            - run: npm install
+      - setup_remote_docker:
+          docker_layer_caching: false
+      - run:
+          name: Install docker-compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.12.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+            chmod +x /usr/local/bin/docker-compose
+      - run:
+          name: Creating mock env file
+          command: touch paramedics-react.env
+      - run:
+          name: Run Docker compose
+          command: |
+            set -x
+            docker-compose -f docker-compose-dev.yaml up -d
+      - persist_to_workspace:
+          root: .
+          paths:
+            - .
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+
+  lint:
+    docker:
+      - image: uwbpparamedics/uw-blueprint-paramedics-react:0.0.2
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ checksum "package.json" }}
+            - v1-dependencies-
+      - attach_workspace:
+          at: .
+      - run:
+          name: Lint
+          command: |
+            npm install
+            npm run lint
+
 workflows:
-  node-tests:
+  version: 2
+  build-lint:
     jobs:
-      - node/test
+      - install-dev
+      - install-prod
+      - build-prod:
+          requires:
+            - install-prod
+      - build-dev:
+          requires:
+            - install-dev
+      - lint:
+          requires:
+            - build-dev
```

if we can remove the `dev-lint` dependency for `dev-lint` that would be ideal - then we could have 3 parallels jobs running and it would be quicker overall
```suggestion
      - prod-build
      - dev-build
      - dev-lint:
        requires:
          - dev-build
```